### PR TITLE
add setup script and update package scripts

### PR DESCRIPTION
**Pull request by sokra**
**What kind of change does this PR introduce?**
feature

**Did you add tests for your changes?**
no

**If relevant, link to documentation update:**
no

**Summary**
revive #6880

**Does this PR introduce a breaking change?**
no

**Other information**

### CODE REVIEW

File: setup/setup.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+function installYarnAsync() {
+	return exec("npm", ["install", "-g", "yarn"], "Install yarn");
+}
+
+function exec(command, args, description) {
+	console.log(`Setup: ${description}`);
+	return new Promise((resolve, reject) => {
+		let cp = require("child_process").spawn(command, args, {
```
Comment (evilebottnawi) [inline setup/setup.js]: I would recommend using `execa`. It is already promised, it is more stable on `windows/freebsd`, it is have simple api and increased child_proccess buffer and etc

Comment (sokra) [reply setup/setup.js]: Good idea, but sadly we can't use any dependency, since the setup script runs before dependencies are installed.

**Comment by sokra** [comment]
Thanks to @Legends for the initial work

File: setup/setup.js (hunk 1 above)
Comment (Legends) [reply setup/setup.js]: Exactly, I was just looking for `execa` in the Node API, but it's a separate package.

**Comment by Legends** [comment]
@sokra The code looks a bit different now, but much better I guess :D, thanks for the improvement!

**Pull request merged**